services/src/main/java/org/keycloak/protocol/saml/SamlService.java

```diff
@@ -307,7 +307,7 @@ protected Response loginRequest(String relayState, AuthnRequestType requestAbstr
                 } else {
                     redirect = client.getAttribute(SamlProtocol.SAML_ASSERTION_CONSUMER_URL_REDIRECT_ATTRIBUTE);
                 }
-                if (redirect == null) {
+                if (redirect == null || redirect.isEmpty()) {
```

Comment:
Should we use `trim()` here? Similarly to this: https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/protocol/saml/SamlService.java#L726

If yes it should be fixed also here: https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/protocol/saml/installation/SamlSPDescriptorClientInstallation.java#L59

BTW: there is also a copy&paste bug. Should we fix it here or in separate Jira?

Reply:
I think it doesn't hurt, changing to use trim(). Fixing the copy&paste bug as well.